File: 942. DI String Match.js

var diStringMatch = function(s) {
let cnt1 = -1, cnt2 = s.length + 1, res = [];
for (let i = 0; i <= s.length; i++) {
    if (s[i] === "I") {
        cnt1++;
        res.push(cnt1);
    } else {
        cnt2--;
        res.push(cnt2);
    }
}
return res;
};

File: 942. DI String Match.rs
impl Solution {
    pub fn di_string_match(s: String) -> Vec<i32> {
    let mut cnt1: i32 = -1;
    let mut cnt2: i32 = s.len() as i32 + 1;
    let mut res: Vec<i32> = Vec::new();
    for i in 0..=s.len() {
        if s.chars().nth(i) == Some('I') {
            cnt1 += 1;
            res.push(cnt1);
        } else {
            cnt2 -= 1;
            res.push(cnt2);
        }
    }
    res 
    }
}
